HyperPlay-Gaming/hyperplay-desktop-client
# [UX] Improvements to Update modal on the Store

Since we got some feedback from users and devs that are not aware of the auto update and thinks that they always need to download the update manually.

It adds 2 new IPC calls that will be used from the store to check the current update status that can be `updating`, `updated` or `idle` (fallback).
Depending on the state we show a different message and action on the store.
If updating, it is possible to close the modal.
If has updated, a button to restart the client is shown.
If idle, we go back to the current behavior of showing the URL to download.

Also, re-enabled the `autoInstallOnAppQuit` option. I think was `false` by mistake.

Store changes: https://github.com/HyperPlay-Gaming/hyperplay-store/pull/223

### How to test

- Download and install the Windows build here from this PR.
- Open HyperPlay and the Update Modal on the Store show should the `Updating` message.
- On the HyperPlay logs screen you can see the download progress messages to confirm the app is being downloaded.
- When asked if you want to restart to conclude the Update, select NO.
- On the Store now the Update Modal should tell the update was completed and show a button to restart the client and apply the update.
- If the client is not updating in the background (test on mac and linux that doesn't have auto update), it should show the current modal we have now.
- Limitations: Messages are not updated automatically on the Store, only when navigating back and forth there.


### Screenshots
![image](https://github.com/user-attachments/assets/922bb914-ffd6-4f35-b87c-a4ab27b4d9b7)
![image](https://github.com/user-attachments/assets/f9779c71-6a39-4ba9-bb14-aa0c8592c2ab)

---

Use the following Checklist if you have changed something on the Backend or Frontend:
